src/bp/core/repositories/sessions.ts

```diff
@@ -137,12 +167,33 @@ export class KnexSessionRepository implements SessionRepository {
     }
 
     await req
+
+    if (this.cache.has(session.id)) {
+      this.cache.del(session.id)
+    } else if (this.ownershipCache.has(session.id)) {
+      this.changeSessionOwnership(session.id, undefined)
+    }
   }
 
   async delete(id: string): Promise<void> {
     await this.database
       .knex(this.tableName)
       .where({ id })
       .del()
+
+    if (this.cache.has(id)) {
+      this.cache.del(id)
+    } else if (this.ownershipCache.has(id)) {
+      this.changeSessionOwnership(id, undefined)
+    }
```

Comment:
It wouldn't be to hard to batch deletes as well if we wanted (I don't it would have much of benefit though compared to batched updates)